src/renderer/components/watch-video-playlist/watch-video-playlist.js

```diff
@@ -232,6 +216,18 @@ export default defineComponent({
     }
   },
   methods: {
+    findIndexOfCurrentVideoInPlaylist: function (playlist) {
+      return playlist.findIndex((item) => {
+        if (item.playlistItemId != null && (this.playlistItemId != null || this.prevVideoBeforeDeletion?.playlistItemId != null)) {
+          return item.playlistItemId === this.playlistItemId || item.playlistItemId === this.prevVideoBeforeDeletion?.playlistItemId
+        } else if (item.videoId != null) {
+          return item.videoId === this.videoId || item.videoId === this.prevVideoBeforeDeletion?.videoId
+        } else {
+          return item.id === this.videoId || item.id === this.prevVideoBeforeDeletion?.videoId
+        }
+      })
+    },
```

Comment:
Caching these 3 values, that are accessed multiple times for every iteration, might help a bit performance wise.
```suggestion
    findIndexOfCurrentVideoInPlaylist: function (playlist) {
      const playlistItemId = this.playlistItemId
      const prevVideoBeforeDeletion = this.prevVideoBeforeDeletion
      const videoId = this.videoId

      return playlist.findIndex((item) => {
        if (item.playlistItemId != null && (playlistItemId != null || prevVideoBeforeDeletion?.playlistItemId != null)) {
          return item.playlistItemId === playlistItemId || item.playlistItemId === prevVideoBeforeDeletion?.playlistItemId
        } else if (item.videoId != null) {
          return item.videoId === videoId || item.videoId === prevVideoBeforeDeletion?.videoId
        } else {
          return item.id === videoId || item.id === prevVideoBeforeDeletion?.videoId
        }
      })
    },
```

Reply:
Thanks! Two notes on this while we're looking at it:
- Do you know if the non-null checks have any effect on the performance? I'm not sure why they're there honestly. 
- I also don't fully know what `item.id` is versus `item.videoId` and what scenario it applies to where the former would be present when the latter isn't.

Reply:
> Do you know if the non-null checks have any effect on the performance? I'm not sure why they're there honestly.

AFAIK because it's both a null and undefined check instead of just being one of them (notice that it only has one equals sign after the exclamation point) it still has to do type coercion, so we could probably convert them to just being truthy checks, as they will always either be strings or missing.

> I also don't fully know what item.id is versus item.videoId and what scenario it applies to where the former would be present when the latter isn't.

I haven't looked into it, but I presume it has something to do with us using the same component for playlists from 3 different sources (local API, Invidious API and users playlists), so they are probably mutually exclusive.